src/NuGet.Clients/PackageManagement.UI/Feeds/IPackageFeed.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.PackageManagement.UI
+{
+    /// <summary>
+    /// Package feed abstraction providing services of package enumeration with search criteria.
+    /// Supports pagination and background processing.
+    /// </summary>
+    internal interface IPackageFeed
```

Comment:
Give this a name specific to search, it sounds too much like the extremely generic IPackageRepository of nuget.core v2.